ci/analyze.sh

```diff
@@ -33,108 +33,43 @@ FLUTTER_DIR="$SRC_DIR/flutter"
-  fi
-  return 0
-)
 
-echo "Analyzing dart:ui library..."
-analyze \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$SRC_DIR/out/host_debug_unopt/gen/dart-pkg/sky_engine/lib/ui/ui.dart"
-
-echo "Analyzing spirv library..."
-analyze \
-  --packages="$FLUTTER_DIR/lib/spirv/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/lib/spirv"
-
-echo "Analyzing ci/"
-analyze \
-  --packages="$FLUTTER_DIR/ci/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/ci"
-
-echo "Analyzing flutter_frontend_server..."
-analyze \
-  --packages="$FLUTTER_DIR/flutter_frontend_server/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/flutter_frontend_server"
-
-echo "Analyzing tools/licenses..."
-analyze \
-  --packages="$FLUTTER_DIR/tools/licenses/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/tools/licenses/analysis_options.yaml" \
-  "$FLUTTER_DIR/tools/licenses"
-
-echo "Analyzing testing/litetest"
-analyze \
-  --packages="$FLUTTER_DIR/testing/litetest/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/litetest"
-
-echo "Analyzing testing/benchmark"
-analyze \
-  --packages="$FLUTTER_DIR/testing/benchmark/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/benchmark"
-
-echo "Analyzing testing/smoke_test_failure"
-analyze \
-  --packages="$FLUTTER_DIR/testing/smoke_test_failure/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/smoke_test_failure"
-
-echo "Analyzing testing/dart..."
-analyze \
-  --packages="$FLUTTER_DIR/testing/dart/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/dart"
-
-echo "Analyzing testing/scenario_app..."
-analyze \
-  --packages="$FLUTTER_DIR/testing/scenario_app/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/scenario_app"
-
-echo "Analyzing testing/symbols..."
-analyze \
-  --packages="$FLUTTER_DIR/testing/symbols/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/testing/symbols"
-
-echo "Analyzing githooks..."
-analyze \
-  --packages="$FLUTTER_DIR/tools/githooks/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/tools/githooks"
-
-echo "Analyzing tools/clang_tidy"
-analyze \
-  --packages="$FLUTTER_DIR/tools/clang_tidy/.dart_tool/package_config.json" \
-  --options "$FLUTTER_DIR/analysis_options.yaml" \
-  "$FLUTTER_DIR/tools/clang_tidy"
+echo "Using dart from $DART_BIN"
+"$DART" --version
+echo ""
+
+# Analyze dart:ui.
+dart analyze "$FLUTTER_DIR/lib/ui"
+
+# Analyze Flutter web's dart:ui.
+(cd "$FLUTTER_DIR/lib/web_ui"; dart pub get)
```

Comment:
Can we somehow avoid running `pub get` from here? I know we run it below for `web_sdk`. We're trying to move to doing `pub get --offline` from gclient sync to avoid random network connection failures messing up various build steps.

Reply:
(I'm taking a quick look to see if I can migrate the pubspec there to work offline)

Reply:
I can remove this from the PR? It wasn't validated here before.

I can also switch to using `--offline` here - I'm not sure how set up the pub cache is at this point. We could also not run `pub get`, and rely on some other build step running it. I'm not familiar enough w/ the build of this repo to say.

I would like to work towards being able to analyze the dart code in this repo on its own - without it having to be a full gclient checkout and having run a large ninja build. That would enable us to do more pre-submit testing of the dart code in this repo from the dart-lanbg/sdk CI system.

Reply:
The pubspec has some non-trivial deps in it. It'd be good to migrate but will likely take some time.

However, we also run analysis on that target in `//lib/web_ui/dev/web_engine_analysis.sh`, so it seems like we probably shouldn't be doing it here - at least until it can avoid doing a pub get.

Reply:
You don't need to do a full ninja build to analyze dart:ui - you can just build the `sky_engine` target, which is a pretty small target that mainly just copies files around.

Doing it without a gclient sync is a tougher ask - we really need to do that to make sure we're getting the right `dart:*` libraries and package dependencies we have from the SDK.

Reply:
OK, I removed the web_ui validation from this PR.

Reply:
@dnfield - wrt the sky_engine target - is that just running `ninja -v -C out/host_debug_unopt flutter/sky/packages/sky_engine`? Or something else?

Reply:
That's correct